lib/index.js

```diff
@@ -1,25 +1,60 @@
-import {EventEmitter} from 'events';
+// @flow
+import EventEmitter from 'events';
 import arrayUniq from 'array-uniq';
 import arrayDiffer from 'array-differ';
-import objectAssign from 'object-assign';
-
-export default class Pageres {
+import * as util from './util';
+
+export type Options = {
+	delay?: number;
+	timeout?: number;
+	crop?: boolean;
+	css?: string;
+	cookies?: Array<string> | Object; // TODO how can I use `[cookies: string]: string` here instead of `Object`?
+	filename?: string;
+	selector?: string;
+	hide?: Array<string>;
+	username?: string;
+	password?: string;
+	scale?: number;
+	format?: string;
+	userAgent?: string;
+	[headers: string]: string; // TODO: is this valid? Couldn't use `?` here.
+};
+
+type Src = {
+	url: string;
+	sizes: Array<string>;
+	options: Options
+};
+
+export default class Pageres extends EventEmitter {
 	/**
 	 * Initialize a new Pageres
 	 *
 	 * @param {Object} options
 	 * @api public
 	 */
 
-	constructor(options) {
-		this.options = objectAssign({}, options);
+	options: Options;
+	stats: Object;
+	items: Array<stream$Readable>; // TODO: is `stream$Readable` correct?
```

Comment:
In this scenario you could do:

```
import {Readable} from 'stream'
```

But if you were writing a libdef you could not because we don't support having dependencies yet: https://github.com/flowtype/flow-typed/issues/16